stateresolutionv2.go

```diff
@@ -374,49 +374,51 @@ func (r *stateResolverV2) authAndApplyEvents(events []*Event) {
 
 // applyEvents applies the events on top of the partial state.
 func (r *stateResolverV2) applyEvents(events []*Event) {
-	for _, e := range events {
-		event := e
-		switch event.Type() {
+	for _, event := range events {
+		st, sk := event.Type(), event.StateKey()
+		switch st {
 		case MRoomCreate:
 			// Room creation events are only valid with an empty state key.
-			if event.StateKey() == nil || *event.StateKey() == "" {
+			if sk == nil || *sk == "" {
 				r.resolvedCreate = event
 			}
 		case MRoomPowerLevels:
 			// Power level events are only valid with an empty state key.
-			if event.StateKey() == nil || *event.StateKey() == "" {
+			if sk == nil || *sk == "" {
```

Comment:
This surely makes it possible for an `m.room.power_levels` event with no state key (aka a message event) pass these checks? We should be checking `if sk != nil && *sk == ""`

Reply:
Whoops, good catch. Thanks.